src/compiler/diagnosticMessages.json

```diff
@@ -5013,5 +5013,41 @@
     "JSX expressions may not use the comma operator. Did you mean to write an array?": {
         "category": "Error",
         "code": 18007
+    },
+    "Private names are not allowed in variable declarations.": {
+        "category": "Error",
+        "code": 18008
+    },
+    "Private names cannot be used as parameters": {
+        "category": "Error",
+        "code": 18009
+    },
+    "Accessibility modifiers cannot be used with private names.": {
+        "category": "Error",
+        "code": 18010
+    },
+    "The operand of a delete operator cannot be a private name.": {
+        "category": "Error",
+        "code": 18011
+    },
+    "'#constructor' is a reserved word.": {
+        "category": "Error",
+        "code": 18012
+    },
+    "Property '{0}' is not accessible outside class '{1}' because it has a private name.": {
+        "category": "Error",
+        "code": 18013
+    },
+    "This usage of '{0}' refers to the private member declared in its enclosing class. While type '{1}' has a private member with the same spelling, its declaration and accessibility are distinct.": {
+        "category": "Error",
+        "code": 18014
+    },
+    "Property '{0}' is missing in type '{1}'. While type '{1}' has a private member with the same spelling, its declaration and accessibility are distinct.": {

```

Comment:
I don't think the "missing" distinction is actually helpful. What about the following?

```
Property '{0}' in type '{1}' refers to a different member that cannot be accessed from within type '{2}'.
```